File: leetcode/0881-boats-to-save-people/numRescueBoats_test.go
package leetcode

import (
	"fmt"
	"path/filepath"
	"testing"

	"github.com/edram/leetcode/util"
)

type NumRescueBoatsCase struct {
	Input struct {
		People []int
		Limit  int
	}

	Output int
}

func TestNumRescueBoats(t *testing.T) {
	var casePath, _ = filepath.Abs("./cases.json")
	payload, err := util.LoadCases[[]NumRescueBoatsCase](casePath)
	if err != nil {
		panic(err)
	}

	for i, testCase := range *payload {
		fmt.Printf("正在执行第 %d 个测试用例\n", i+1)
		fmt.Printf("string: %v ==> %v \n", testCase.Input, testCase.Output)

		result := numRescueBoats(testCase.Input.People, testCase.Input.Limit)
		if result != testCase.Output {
			t.Errorf("期望 %v, 结果 %v", testCase.Output, result)
		}
	}
}

File: leetcode/0881-boats-to-save-people/numRescueBoats_test.rs
use std::{env::current_dir, path::Path};

use serde::Deserialize;

use crate::{util::load_cases, Solution};

#[derive(Deserialize, Debug)]
struct Input {
    people: Vec<i32>,
    limit: i32,
}

#[derive(Deserialize, Debug)]
struct Case {
    input: Input,
    output: i32,
}

#[test]
fn test_boats_to_save_people() {
    let current_path = current_dir().unwrap().join(file!());
    let dir = current_path.parent().unwrap();

    let cases: Vec<Case> = load_cases(Path::new(&dir.join("./cases.json")));

    for case in &cases {
        assert_eq!(
            Solution::num_rescue_boats(case.input.people.clone(), case.input.limit),
            case.output
        );
    }
}
